tests/testthat/error/test-conditions.txt

```diff
@@ -0,0 +1,20 @@
+
+can override arg in OOB conditions
+==================================
+
+> with_subscript_data(vec_slice(set_names(letters), "foo"), NULL)
```

Comment:
These might be easier to understand if you just called `stop_subscript_oob_name()` directly?

Reply:
hmm I think my goal with these tests was to make sure that `vec_slice()` throws the right errors even when we change the internals of the errors. For instance `stop_subscript_oob_name()` doesn't exist anymore in the later PRs.

Reply:
Yeah. I'm not sure how to improve it, but when reading through this file, I didn't really get any sense of what was being customised in the error message.